addon/-private/system/model/internal-model.js

```diff
@@ -857,44 +982,40 @@ InternalModel.prototype = {
         // this attributes. We never override this value when merging
         // updates from the backend so we should not sent a change
         // notification if the server value differs from the original.
-        if (this._attributes[key] !== undefined) {
+        if (attrs[key] !== undefined) {
           continue;
         }
 
-        if (!Ember.isEqual(original[key], value)) {
+        if (!isEqual(original[key], value)) {
           changedKeys.push(key);
         }
       }
     }
 
     return changedKeys;
-  },
+  }
 
   toString() {
-    if (this.record) {
-      return this.record.toString();
-    } else {
-      return `<${this.modelName}:${this.id}>`;
-    }
-  },
+    return `<${this.modelName}:${this.id}>`;
+  }
 
-  referenceFor(type, name) {
+  referenceFor(modelClass, name) {
```

Comment:
whoops, will change that back. I think maybe we should consistently refer to this as `kind` internally? I noticed this confusion in another spot too.